common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -965,24 +965,24 @@ private static boolean maybeSuperUser0() {
         return "root".equals(username) || "toor".equals(username);
     }
 
-    private static boolean hasUnsafe0() {
+    private static Throwable unsafeUnavailabilityCause0() {
         if (isAndroid()) {
             logger.debug("sun.misc.Unsafe: unavailable (Android)");
-            return false;
+            return new UnsupportedOperationException("sun.misc.Unsafe: unavailable (Android)");
         }
 
         if (PlatformDependent0.isExplicitNoUnsafe()) {
```

Comment:
this conditional is already taken into account in `PlatformDependent0` when setting `PlatformDependent0.getUnsafeUnavailabilityCause()` ... can we just let `PlatformDependent0` be responsible for enforcing this conditional? For example:

```java
Throwable cause = PlatformDependent0.getUnsafeUnavailabilityCause();
if (cause != null) {
  return cause;
}
...
```

Reply:
@carl-mastrangelo ^^

Reply:
Sorry for the delay, done!